blocks/search-results/search-results.js

```diff
@@ -0,0 +1,122 @@
+import { createCardBlock } from '../card/card.js';
+
+export default async function decorate(block) {
+  block.innerHTML = '';
+  const searchTerm = new URLSearchParams(window.location.search).get('q');
+  block.classList.add('card-container', 'three-columns');
+  if (searchTerm) {
+    const elementHeading = block.parentNode.parentNode.parentNode.querySelector('.section.search-page-heading');
+    const textNode = document.createElement('h1');
+    textNode.textContent = searchTerm;
+    elementHeading.append(textNode);
+    const mainDiv = document.createElement('div');
+    mainDiv.innerHTML = '<div class="results-loading-spinner"><div>';
+    const resultsDiv = document.createElement('div');
+    resultsDiv.classList.add('results-div');
+    mainDiv.append(resultsDiv);
+    block.append(mainDiv);
+    const tokenizedSearchWords = searchItems(searchTerm);
+    // noinspection ES6MissingAwait
+    loadResults(tokenizedSearchWords, resultsDiv);
+  } else {
+    window.location.href = '/';
+  }
+}
+
+function searchItems(searchTerm) {
+  const tokenizedSearchWords = searchTerm.split(' ');
+  if (tokenizedSearchWords.length > 1) tokenizedSearchWords.unshift(searchTerm);
+  return tokenizedSearchWords;
+}
+
+async function loadResults(tokenizedSearchWords, resultsDiv) {
+  const allData = await fetch(`${window.location.origin}/articles.json?sheet=full`);
+  const jsonData = await allData.json();
+  const matches = filterMatches(tokenizedSearchWords, jsonData);
+  resultsDiv.parentNode.querySelector('.results-loading-spinner').remove();
+  if (matches.length === 0) {
+    noResults(resultsDiv);
+  } else {
+    createCards(matches, resultsDiv);
+  }
+}
+
+function filterMatches(tokenizedSearchWords, jsonData) {
+  let allData = [];
+  tokenizedSearchWords.forEach((element) => {
+    const match = jsonData.data.filter((entry) => (
+      entry.title
+      + entry.content
+      + entry.path
+      + entry.authors
+      + entry.collections
+      + entry.section
+      + entry.categories
+    )
+      .toLowerCase()
+      .includes(element.toLowerCase()));
+    if (match) { allData = allData.concat(match); }
+  });
+  return uniqueMatches(allData);
+}
```

Comment:
avoid generic terms like `data` and `element`, it adds no value. `matches` (plural) makes it clear that it's an array, not a single result. 
For ... see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax 
```suggestion
function filterMatches(tokenizedSearchWords, jsonData) {
  const allMatches = [];
  tokenizedSearchWords.forEach((searchTerm) => {
    const matches = jsonData.data.filter((entry) => (
      entry.title
      + entry.content
      + entry.path
      + entry.authors
      + entry.collections
      + entry.section
      + entry.categories
    )
      .toLowerCase()
      .includes(searchTerm.toLowerCase()));
    allMatches.push(...matches);
  });
  return uniqueMatches(allMatches);
}
```

Reply:
Fixed